apps/ui/src/views/User.vue

```diff
@@ -1,38 +1,232 @@
 <script setup lang="ts">
-import { shortenAddress, _n } from '@/helpers/utils';
+import {
+  shortenAddress,
+  autoLinkText,
+  _n,
+  _p,
+  getCacheHash,
+  isValidAddress,
+  getSocialNetworksLink
+} from '@/helpers/utils';
+import { enabledNetworks, getNetwork } from '@/networks';
+import { UserActivity, Space, User } from '@/types';
+import { getNames } from '@/helpers/stamp';
 
 const route = useRoute();
 const usersStore = useUsersStore();
+const spacesStore = useSpacesStore();
+const { web3 } = useWeb3();
 const { setTitle } = useTitle();
+const { copy, copied } = useClipboard();
 
-const id = route.params.id as string;
-const user = computed(() => usersStore.getUser(id));
+const activities = ref<
+  (UserActivity & { space: Space; proposal_percentage: number; vote_percentage: number })[]
+>([]);
+const loadingActivities = ref(false);
+const modalOpenEditUser = ref(false);
+const loaded = ref(false);
 
-onMounted(() => usersStore.fetchUser(id));
+const id = computed(() => route.params.id as string);
 
-watchEffect(() => setTitle(`${id} user profile`));
+const user = computedAsync(
+  async () =>
+    usersStore.getUser(id.value) ||
+    (isValidAddress(id.value)
+      ? ({
+          id: id.value,
+          name: (await getNames([id.value]))?.[id.value]
+        } as User)
+      : null)
+);
+
+const socials = computed(() => getSocialNetworksLink(user.value));
+
+const shareMsg = computed(() => encodeURIComponent(window.location.href));
+
+const cb = computed(() => getCacheHash(user.value?.avatar));
+
+async function loadActivities(userId: string) {
+  loadingActivities.value = true;
+
+  try {
+    const results = await Promise.all(
+      enabledNetworks.map(networkId => getNetwork(networkId).api.loadUserActivities(userId))
+    );
+
+    const aggregatedActivities = results
+      .flat()
+      .sort((a, b) => b.proposal_count - a.proposal_count || b.vote_count - a.vote_count);
+
+    await spacesStore.fetchSpaces(
+      aggregatedActivities
+        .map(activity => activity.spaceId)
+        .filter(id => !spacesStore.spacesMap.has(id))
+    );
+
+    const totalProposals = aggregatedActivities
+      .map(activity => activity.proposal_count)
+      .reduce((a, b) => a + b, 0);
+
+    const totalVotes = aggregatedActivities
+      .map(activity => activity.vote_count)
+      .reduce((a, b) => a + b, 0);
```

Comment:
nit. this goes through `aggregatedActivities` 4 times total, but can be done in one pass with similarly clean approach:
```js
const [totalProposals, totalVotes] = aggregatedActivities.reduce(
  ([totalProposals, totalVotes], activity) => [
    totalProposals + activity.proposal_count,
    totalVotes + activity.vote_count
  ],
  [0, 0]
);
```
Alternatively I'd prefer to skip map and just do:
```js
const totalProposals = aggregatedActivities.reduce((a, b) => a + b.proposal_count, 0);
const totalVotes = aggregatedActivities.reduce((a, b) => a + b.vote_count, 0);
```